scripts/quests/jeuno/Scattered_into_Shadow.lua

```diff
@@ -0,0 +1,204 @@
+                    end
+                end,
+
+            },
+            onEventFinish =
+            {
+                [143] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                        npcUtil.giveKeyItem(player, { xi.ki.AQUAFLORA1, xi.ki.AQUAFLORA2, xi.ki.AQUAFLORA3 })
+                    end
+                end,
+
+                [141] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                        npcUtil.giveKeyItem(player, { xi.ki.AQUAFLORA1, xi.ki.AQUAFLORA2, xi.ki.AQUAFLORA3 })
+                    else
+                        quest:setVar(player, 'Prog', 1)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.FEIYIN] =
+        {
+            ['Underground_Pool'] =
+            {
+                onTrigger = function(player, npc)
+                    local offset = npc:getID() - feiyinID.npc.UNDERGROUND_POOL_OFFSET
+
+                    if
+                        offset == 0 and
+                        player:hasKeyItem(xi.ki.AQUAFLORA2)
+                    then
+                        return quest:progressEvent(20)
+                    elseif
+                        offset == 1 and
+                        quest:getVar(player, 'Prog') == 2
+                    then
+                        return quest:progressEvent(18)
+                    elseif
+                        offset == 1 and
+                        player:hasKeyItem(xi.ki.AQUAFLORA3) and
+                        not GetMobByID(feiyinID.mob.DABOTZS_GHOST):isSpawned() and
+                        npcUtil.popFromQM(player, npc, feiyinID.mob.DABOTZS_GHOST, { claim = true, hide = 0 })
+                    then
+                        return quest:noAction()
+                    elseif
+                        offset == 2 and
+                        player:hasKeyItem(xi.ki.AQUAFLORA1)
+                    then
+                        return quest:progressEvent(21)
+                    end
+                end,
+            },
+
+            ['Dabotzs_Ghost'] =
+            {
+                onMobDeath = function(mob, player, optParams)
+                    if player:hasKeyItem(xi.ki.AQUAFLORA3) then
+                        quest:setVar(player, 'Prog', 2)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [18] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.AQUAFLORA3)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+
+                [20] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.AQUAFLORA2)
```

Comment:
For these various add/del key items, is this messaged in the CS that's just finished, or should these handle their own messaging? What does the cap show?

Reply:
The CS/event for each of the flora removals delivers the message to the player, however I missed the message when giving the player the aquaflora from Brutus. Added!